packages/starlight-ghostcms/src/integrations/starlight-ghostcms.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    createResolver,
+    defineIntegration,
+} from "astro-integration-kit";
+import { corePlugins } from "astro-integration-kit/plugins";
+import { z } from "astro/zod";
+import { type StarlightGhostConfig } from "../schemas/config";
+
+export default defineIntegration({
+    name: "@matthiesenxyz/starlight-ghostcms",
+    optionsSchema: z.custom<StarlightGhostConfig>(),
+    plugins: [...corePlugins],
+    setup({ options }) {
+        const { resolve } = createResolver(import.meta.url);
+
+        return {
+            "astro:config:setup": ({ 
+                watchIntegration, 
+                addVirtualImports, 
+                injectRoute 
+            }) => {
+                watchIntegration(resolve());
+
+                addVirtualImports({
+                    'virtual:starlight-ghostcms/config': `export default ${JSON.stringify(options)}`,
+                });
+
+                const makeRoute = (endpoint: string, entrypoint: string) => {
+                    injectRoute({
+                        pattern: `/${endpoint}`,
+                        entrypoint: resolve(`../routes/${entrypoint}`),
+                        prerender: true,
+                    });
+                };
+                const sanitisedRoute = options.route
```

Comment:
Guhuhuhuhu it's so beautiful 😭 It's the best fucking line in this PR hands down. I can't believe how beautiful it is!